diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2719b40..314b4e2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,13 +25,18 @@ add_definitions(-DMBED_CONF_MBED_CLIENT_SN_COAP_DUPLICATION_MAX_MSGS_COUNT=4)
 
 project(mbedCloudClient)
 
+#!!! 
+# Patch: some of the excessive include_directories commented out in order to make the includes ("-I...") length 
+# in CFLAGS as short as possible. Without this patch we get compilation failure "execv: argument too long" 
+# due to excessive include directories. This bug appeared in integration of mbed-cloud-client 1.4.0 into MBL.
+
 # mbed-cloud-client
 
 ADD_GLOBALDIR( ${CMAKE_CURRENT_SOURCE_DIR})
 ADD_GLOBALDIR( ${CMAKE_CURRENT_SOURCE_DIR}/source)
 ADD_GLOBALDIR( ${CMAKE_CURRENT_SOURCE_DIR}/source/include)
 ADD_GLOBALDIR( ${CMAKE_CURRENT_SOURCE_DIR}/mbed-cloud-client)
-ADD_GLOBALDIR( ${CMAKE_CURRENT_SOURCE_DIR}/mbed-client)
+# ADD_GLOBALDIR( ${CMAKE_CURRENT_SOURCE_DIR}/mbed-client)
 
 # mbed-client
 
@@ -47,7 +52,7 @@ ADD_GLOBALDIR(${MBED_CLIENT_SOURCE_DIR}/mbed-client-c/source/include)
 ADD_GLOBALDIR(${MBED_CLIENT_SOURCE_DIR}/mbed-client-classic)
 ADD_GLOBALDIR(${MBED_CLIENT_SOURCE_DIR}/mbed-client-classic/mbed-client-classic)
 ADD_GLOBALDIR(${MBED_CLIENT_SOURCE_DIR}/mbed-client-mbed-tls)
-ADD_GLOBALDIR(${MBED_CLIENT_SOURCE_DIR}/mbed-client-mbed-tls/mbed-client-mbedtls)
+# ADD_GLOBALDIR(${MBED_CLIENT_SOURCE_DIR}/mbed-client-mbed-tls/mbed-client-mbedtls)
 
 # pal headers
 
@@ -61,21 +66,21 @@ ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-coap)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-coap/mbed-coap)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-coap/source/include)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-trace)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-trace/mbed-trace)
-ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-client-randlib)
+# ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-client-randlib)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-client-randlib/mbed-client-randlib)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/mbed-client-randlib/mbed-client-randlib/platform)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/nanostack-libservice)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/nanostack-libservice/mbed-client-libservice)
-ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/nanostack-libservice/mbed-client-libservice/platform)
-ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/sal-stack-nanostack-eventloop)
+# ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/nanostack-libservice/mbed-client-libservice/platform)
+# ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/sal-stack-nanostack-eventloop)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/sal-stack-nanostack-eventloop/nanostack-event-loop)
 ADD_GLOBALDIR(${CMAKE_CURRENT_SOURCE_DIR}/ns-hal-pal)
 
 # factory-client
 
 SET(FACTORY_CLIENT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/factory-configurator-client)
-ADD_GLOBALDIR(${FACTORY_CLIENT_SOURCE_DIR}/factory-configurator-client)
+# ADD_GLOBALDIR(${FACTORY_CLIENT_SOURCE_DIR}/factory-configurator-client)
 
 SET(FCC_MODULES
     ftcd-comm-base
@@ -101,27 +106,27 @@ SET(FCC_MODULES
 #ENDFOREACH()
 
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/factory-configurator-client/source/include)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/factory-configurator-client/factory-configurator-client)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fc_protocol_handler/fc_protocol_handler)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fc_protocol_handler/source/include)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/secure_store)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/secure_store/secure_store)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/storage)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/factory-configurator-client/factory-configurator-client)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fc_protocol_handler/fc_protocol_handler)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fc_protocol_handler/source/include)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/secure_store)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/secure_store/secure_store)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/storage)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/storage/storage)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/key-config-manager)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/key-config-manager/source/include)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/utils)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/utils)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/utils/utils)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/logger)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/logger)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/logger/logger)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/mbed-client-esfs/source/include)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/mbed-client-esfs/source-pal/api)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/mbed-client-esfs/source-pal/linux)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/crypto-service/crypto-service)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/crypto-service/source/include)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/mbed-client-esfs/source/include)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/mbed-client-esfs/source-pal/api)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/mbed-client-esfs/source-pal/linux)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/crypto-service/crypto-service)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/crypto-service/source/include)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/secsrv-cbor/secsrv-cbor)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fcc-bundle-handler/fcc-bundle-handler)
-include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fcc-bundle-handler/source/include)
+# include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fcc-bundle-handler/source/include)
 include_directories(${FACTORY_CLIENT_SOURCE_DIR}/fcc-output-info-handler/fcc-output-info-handler)
 
 # Certificate Enrollment Client
