diff --git a/mbed-client-pal/Source/Port/Reference-Impl/OS_Specific/Linux/RTOS/pal_plat_rtos.c b/mbed-client-pal/Source/Port/Reference-Impl/OS_Specific/Linux/RTOS/pal_plat_rtos.c
index 5826457..b1d0855 100644
--- a/mbed-client-pal/Source/Port/Reference-Impl/OS_Specific/Linux/RTOS/pal_plat_rtos.c
+++ b/mbed-client-pal/Source/Port/Reference-Impl/OS_Specific/Linux/RTOS/pal_plat_rtos.c
@@ -230,19 +230,27 @@ palStatus_t pal_plat_osThreadCreate(palThreadFuncPtr function, void* funcArgumen
     int err = pthread_attr_init(&attr);
     if (0 != err)
     {
+        PAL_LOG_ERR("pthread_attr_init failed: %s\n", strerror(err));
         goto finish;
     }
     ptrAttr = &attr;
 
+    if (stackSize < PTHREAD_STACK_MIN)
+    {
+        stackSize = PTHREAD_STACK_MIN;
+    }
+
     err = pthread_attr_setstacksize(ptrAttr, stackSize);
     if (0 != err)
     {
+        PAL_LOG_ERR("pthread_attr_setstacksize failed: %s\n. stackSize=%" PRIu32 "; PTHREAD_STACK_MIN=%zu", strerror(err), stackSize, PTHREAD_STACK_MIN);
         goto finish;
     }
 
     err = pthread_attr_setschedpolicy(ptrAttr, SCHED_RR);
     if (0 != err)
     {
+        PAL_LOG_ERR("pthread_attr_setschedpolicy failed: %s\n", strerror(err));
         goto finish;
     }
 
@@ -250,6 +258,7 @@ palStatus_t pal_plat_osThreadCreate(palThreadFuncPtr function, void* funcArgumen
     err = pthread_attr_setinheritsched(ptrAttr, PTHREAD_EXPLICIT_SCHED);
     if (0 != err)
     {
+        PAL_LOG_ERR("pthread_attr_setinheritsched failed: %s\n", strerror(err));
         goto finish;
     }
 #endif    
@@ -257,6 +266,7 @@ palStatus_t pal_plat_osThreadCreate(palThreadFuncPtr function, void* funcArgumen
     err = pthread_attr_setdetachstate(ptrAttr, PTHREAD_CREATE_DETACHED);
     if (0 != err)
     {
+        PAL_LOG_ERR("pthread_attr_setdetachstate failed: %s\n", strerror(err));
         goto finish;
     }    
     
@@ -264,6 +274,7 @@ palStatus_t pal_plat_osThreadCreate(palThreadFuncPtr function, void* funcArgumen
     err = pthread_attr_setschedparam(ptrAttr, &schedParam);
     if (0 != err)
     {
+        PAL_LOG_ERR("pthread_attr_setschedparam failed: %s\n", strerror(err));
         goto finish;
     }
 
