Index: mbed-cloud-client/mbed-client-pal/Configs/pal_config/Linux/Linux_default.h
===================================================================
--- mbed-cloud-client.orig/mbed-client-pal/Configs/pal_config/Linux/Linux_default.h
+++ mbed-cloud-client/mbed-client-pal/Configs/pal_config/Linux/Linux_default.h
@@ -70,10 +70,18 @@
 // 16KB does not seem to be enough, some tests are failing with it
 #ifndef PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE
     #define PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE (1024 * 24)
+    #if (PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE < PTHREAD_STACK_MIN)
+        #undef PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE
+        #define PAL_NET_TEST_ASYNC_SOCKET_MANAGER_THREAD_STACK_SIZE PTHREAD_STACK_MIN
+    #endif
 #endif
 
 #ifndef PAL_RTOS_HIGH_RES_TIMER_THREAD_STACK_SIZE
     #define PAL_RTOS_HIGH_RES_TIMER_THREAD_STACK_SIZE (1024 * 16)
+    #if (PAL_RTOS_HIGH_RES_TIMER_THREAD_STACK_SIZE < PTHREAD_STACK_MIN)
+        #undef PAL_RTOS_HIGH_RES_TIMER_THREAD_STACK_SIZE
+        #define PAL_RTOS_HIGH_RES_TIMER_THREAD_STACK_SIZE PTHREAD_STACK_MIN
+    #endif
 #endif
 
 #ifndef PAL_FORMAT_CMD_MAX_LENGTH
@@ -125,11 +133,19 @@
 //!< Stack size for thread created when calling pal_getAddressInfoAsync
 #ifndef PAL_NET_ASYNC_DNS_THREAD_STACK_SIZE
     #define PAL_NET_ASYNC_DNS_THREAD_STACK_SIZE (1024 * 32)
+    #if (PAL_NET_ASYNC_DNS_THREAD_STACK_SIZE < PTHREAD_STACK_MIN)
+        #undef PAL_NET_ASYNC_DNS_THREAD_STACK_SIZE
+        #define PAL_NET_ASYNC_DNS_THREAD_STACK_SIZE PTHREAD_STACK_MIN
+    #endif
 #endif
 
 //! Stack size for TRNG noise collecting thread
 #ifndef PAL_NOISE_TRNG_THREAD_STACK_SIZE
     #define PAL_NOISE_TRNG_THREAD_STACK_SIZE (1024 * 32)
+    #if (PAL_NOISE_TRNG_THREAD_STACK_SIZE < PTHREAD_STACK_MIN)
+        #undef PAL_NOISE_TRNG_THREAD_STACK_SIZE
+        #define PAL_NOISE_TRNG_THREAD_STACK_SIZE PTHREAD_STACK_MIN
+    #endif
 #endif
 
 #ifndef PAL_TIMER_SIGNAL
