diff --git a/update-client-hub/modules/pal-filesystem/source/arm_uc_paal_classic_pal.c b/update-client-hub/modules/pal-filesystem/source/arm_uc_paal_classic_pal.c
index 21fcaf75e8e510238ea4d91e614e8dec991dfb81..1d51fd6dd19a62b1c53bd48325a1db7d783cc863 100644
--- a/update-client-hub/modules/pal-filesystem/source/arm_uc_paal_classic_pal.c
+++ b/update-client-hub/modules/pal-filesystem/source/arm_uc_paal_classic_pal.c
@@ -30,7 +30,11 @@
 
 #include <stdio.h>
 
+#ifndef PAL_UPDATE_FIRMWARE_DIR
 #define ARM_UC_FIRMWARE_FOLDER_NAME "firmware"
+#else
+#define ARM_UC_FIRMWARE_FOLDER_NAME PAL_UPDATE_FIRMWARE_DIR
+#endif
 
 /* pointer to external callback handler */
 static ARM_UC_PAAL_UPDATE_SignalEvent_t arm_uc_pal_external_callback = NULL;

