From 6675978d1e7cec5726702ee2356465a9ae8174ef Mon Sep 17 00:00:00 2001
From: Jonathan Haigh <jonathan.haigh@arm.com>
Date: Wed, 26 Jun 2019 08:52:37 +0100
Subject: [PATCH 1/5] Support absolute directories for --build option

---
 api-tests/tools/makefiles/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/api-tests/tools/makefiles/Makefile b/api-tests/tools/makefiles/Makefile
index 1aa790b..6c87c91 100644
--- a/api-tests/tools/makefiles/Makefile
+++ b/api-tests/tools/makefiles/Makefile
@@ -39,7 +39,7 @@ target_cfg:
 	$(BUILD)/platform/${TARGET}/target_database.h target_database ""
 	gcc -D__addr_t_defined -DTARGET_CFG_BUILD $(BUILD)/platform/${TARGET}/targetConfigGen.c -o $(BUILD)/platform/${TARGET}/targetConfigGen \
 	-I$(SOURCE)/val/nspe -I$(SOURCE)/val/common -I$(SOURCE)/platform/targets/${TARGET}/nspe/common
-	./$(BUILD)/platform/${TARGET}/targetConfigGen
+	$(BUILD)/platform/${TARGET}/targetConfigGen
 
 #Read target.cfg and update the addresses in linker script
 gen_linker:
-- 
2.7.4

