diff --git a/cloud-services/mbl-cloud-client/CMakeLists.txt b/cloud-services/mbl-cloud-client/CMakeLists.txt
index 5eca8e2..dddee65 100644
--- a/cloud-services/mbl-cloud-client/CMakeLists.txt
+++ b/cloud-services/mbl-cloud-client/CMakeLists.txt
@@ -42,21 +42,27 @@ add_compile_options(-Wlogical-op)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/source)
 
+
+#!!! 
+# Patch: all excessive include_directories commented out in order to make includes ("-I...") length 
+# in CFLAGS as short as possible. Without this patch we get compilation failure "execv: argument too long" 
+# due to excessive include directories. This bug appeared in integration of mbed-cloud-client 1.4.0 into MBL.
+
 SET(MBED_CLOUD_CLIENT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/mbed-cloud-client)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/mbed-cloud-client)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/source)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/mbed-client)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/mbed-client/mbed-client-c)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/mbed-coap/mbed-coap)
+#include_directories(${CMAKE_CURRENT_SOURCE_DIR}/mbed-cloud-client)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/source)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/mbed-client)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/mbed-client/mbed-client-c)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/mbed-coap/mbed-coap)
 include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/mbed-trace-helper)
 include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/factory-configurator-client)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/factory-configurator-client/factory-configurator-client)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/fcc-output-info-handler/fcc-output-info-handler)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/key-config-manager/key-config-manager)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/update-client-hub)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/update-client-hub/modules/common)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/factory-configurator-client/factory-configurator-client)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/fcc-output-info-handler/fcc-output-info-handler)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/factory-configurator-client/key-config-manager/key-config-manager)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/update-client-hub)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/update-client-hub/modules/common)
 include_directories(${MBED_CLOUD_CLIENT_DIR}/mbed-trace)
-include_directories(${MBED_CLOUD_CLIENT_DIR}/nanostack-libservice/mbed-client-libservice)
+#include_directories(${MBED_CLOUD_CLIENT_DIR}/nanostack-libservice/mbed-client-libservice)
 
 FILE(GLOB MBL_CLOUD_CLIENT_SRC
     "${CMAKE_CURRENT_SOURCE_DIR}/*.c"
@@ -83,3 +89,4 @@ target_link_libraries(mbl-cloud-client mbedCloudClient)
 
 target_link_libraries(mbl-cloud-client pthread)
 target_link_libraries(mbl-cloud-client rt)
+
diff --git a/cloud-services/mbl-cloud-client/pal-platform/Middleware/mbedtls/mbedtls.cmake b/cloud-services/mbl-cloud-client/pal-platform/Middleware/mbedtls/mbedtls.cmake
index 0209546..37d3ce9 100644
--- a/cloud-services/mbl-cloud-client/pal-platform/Middleware/mbedtls/mbedtls.cmake
+++ b/cloud-services/mbl-cloud-client/pal-platform/Middleware/mbedtls/mbedtls.cmake
@@ -19,10 +19,15 @@ SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fomit-frame-pointer")
 SET(ENABLE_PROGRAMS OFF CACHE STRING "Avoid compiling mbedtls programs" )
 SET(ENABLE_TESTING OFF CACHE STRING "Avoid compiling mbedtls tests")
 
+#!!! 
+# Patch: all excessive include_directories commented out in order to make includes ("-I...") length 
+# in CFLAGS as short as possible. Without this patch we get compilation failure "execv: argument too long" 
+# due to excessive include directories. This bug appeared in integration of mbed-cloud-client 1.4.0 into MBL.
+
 include_directories ("${CMAKE_SOURCE_DIR}/pal-platform/Middleware/mbedtls")
 include_directories ("${CMAKE_SOURCE_DIR}/pal-platform/Middleware/mbedtls/mbedtls/include")
 include_directories ("${CMAKE_SOURCE_DIR}/pal-platform/Middleware/mbedtls/mbedtls/include/mbedtls")
-include_directories ("${CMAKE_SOURCE_DIR}/pal-platform/Middleware/mbedtls/mbedtls/port/ksdk")
+# include_directories ("${CMAKE_SOURCE_DIR}/pal-platform/Middleware/mbedtls/mbedtls/port/ksdk")
 # include_directories ("./pal-platform/Middleware/mbedtls/mmcau_2.0.0")
 
 message(status "device = ${PAL_TARGET_DEVICE}")
