diff --git a/CMakeLists.txt b/CMakeLists.txt
index f102b0df465ceba55906e3e318564c30cd3d2ed7..c014f1411a68fe610c99593ed9c6c2169dc024c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -192,6 +192,7 @@ endif()
     "${CMAKE_CURRENT_SOURCE_DIR}/ns-hal-pal/arm_hal_timer.cpp"
     "${FACTORY_CLIENT_SOURCE_DIR}/source/*.c"
     "${FACTORY_CLIENT_SOURCE_DIR}/storage/source/*.c"
+    "${FACTORY_CLIENT_SOURCE_DIR}/mbed-client-esfs/source/sotp/*.c"
     "${FACTORY_CLIENT_SOURCE_DIR}/secure_store/source/*.c"
     "${FACTORY_CLIENT_SOURCE_DIR}/key-config-manager/source/*.c"
     "${FACTORY_CLIENT_SOURCE_DIR}/utils/source/*.c"
diff --git a/mbed-client/mbed-client/m2minterfaceobserver.h b/mbed-client/mbed-client/m2minterfaceobserver.h
index c4257fb8af449d20d3766ffa84c972a723fb0975..36b5a37f21ada2e6ca6d876cdab520c600b06d8b 100644
--- a/mbed-client/mbed-client/m2minterfaceobserver.h
+++ b/mbed-client/mbed-client/m2minterfaceobserver.h
@@ -86,7 +86,7 @@ public:
      * \brief A callback indicating when all bootstrap data has been received.
      * \param security_object, The security object that contains the security information.
      */
-    virtual void bootstrap_data_ready(M2MSecurity *security_object) { };
+    virtual void bootstrap_data_ready(M2MSecurity *) { };
 
 };

