Index: git/include/configs/rpi.h
===================================================================
--- git.orig/include/configs/rpi.h
+++ git/include/configs/rpi.h
@@ -181,7 +181,22 @@
 
 #include <config_distro_bootcmd.h>
 
+#if defined(CONFIG_SILENT_CONSOLE)
+#define SILENT_CONSOLE_VAL 1
+#else
+#define SILENT_CONSOLE_VAL 0
+#endif
+
+#if defined(CONFIG_SILENT_U_BOOT_ONLY)
+#define SILENT_LINUX_CONSOLE_VAL 0
+#else
+#define SILENT_LINUX_CONSOLE_VAL 1
+#endif
+
+
 #define CONFIG_EXTRA_ENV_SETTINGS \
+    "silent=" __stringify(SILENT_CONSOLE_VAL) "\0" \
+    "silent_linux=" __stringify(SILENT_LINUX_CONSOLE_VAL) "\0" \
 	"dhcpuboot=usb start; dhcp u-boot.uimg; bootm\0" \
 	ENV_DEVICE_SETTINGS \
 	ENV_MEM_LAYOUT_SETTINGS \
