From 0205a29cac0395c648fdffaca8a23e57e1051b52 Mon Sep 17 00:00:00 2001
From: nobody <nobody@nowhere.arm.com>
Date: Fri, 1 Nov 2019 15:53:24 +0000
Subject: [PATCH] enable net without net commands kconfig

---
 Kconfig | 3 ---
 1 file changed, 3 deletions(-)

Index: git/Kconfig
===================================================================
--- git.orig/Kconfig
+++ git/Kconfig
@@ -79,15 +79,12 @@ config DISTRO_DEFAULTS
 	select CMDLINE_EDITING
 	select CMD_BOOTI if ARM64
 	select CMD_BOOTZ if ARM && !ARM64
-	select CMD_DHCP if CMD_NET
 	select CMD_ENV_EXISTS
 	select CMD_EXT2
 	select CMD_EXT4
 	select CMD_FAT
 	select CMD_FS_GENERIC
 	select CMD_PART if PARTITIONS
-	select CMD_PING if CMD_NET
-	select CMD_PXE if NET
 	select CMD_SYSBOOT
 	select ENV_VARS_UBOOT_CONFIG
 	select HUSH_PARSER
